configure.ac

```diff
@@ -37,6 +37,14 @@ LIBYAMI_LT_LDFLAGS="-version-number $LIBYAMI_LT_VERSION"
 AC_SUBST(LIBYAMI_LT_VERSION)
 AC_SUBST(LIBYAMI_LT_LDFLAGS)
 
+# check for required macro definitions
+AC_MSG_CHECKING([for ax_pkg_check_modules])
+m4_ifndef([AX_PKG_CHECK_MODULES],
+    [AC_MSG_RESULT([no])
+     m4_defun([AX_PKG_CHECK_MODULES],[])
+     AC_MSG_ERROR([autoconf-archive >= 2015.02.24 is required])],
```

Comment:
I took another look at this, and realized that AX_PKG_CHECK_MODULES is really only useful if we want to split public and private Requires in .pc files.  Turns out that we don't ever use the private Requires part.  Thus, AX_PKG_CHECK_MODULES usage is a bit of overkill and redundant for our situation.

I'll refactor this so that we don't need AX_PKG_CHECK_MODULES.